tensorflow/core/kernels/data/dataset_test_base.h

```diff
@@ -38,9 +38,35 @@ limitations under the License.
 #include "tensorflow/core/platform/types.h"
 #include "tensorflow/core/util/ptr_util.h"
 
+#include "tensorflow/core/lib/io/zlib_compression_options.h"
+#include "tensorflow/core/lib/io/zlib_outputbuffer.h"
+#include "tensorflow/core/platform/env.h"
+
 namespace tensorflow {
 namespace data {
 
+enum CompressionType : int { ZLIB = 0, GZIP = 1, RAW = 2, UNCOMPRESSED = 3 };
+
+// Gets the compression name.
+string CompressionName(CompressionType compression_type);
+
+// Gets the specified zlib compression options according to the compression
+// type. Note that `CompressionType : UNCOMPRESSED` is not supported because
+// `ZlibCompressionOptions` does not have the option for uncompressed data.
+io::ZlibCompressionOptions GetZlibCompressionOptions(
+    CompressionType compression_type);
+
+// Saves the input text into the file with the specified compression.
+Status CreateFile(const string& filename, const std::vector<string>& text,
```

Comment:
why is the input data a vector of string? that seems to be an awkward API

Document what is `input_buffer_size` and `output_buffer_size` for.

Rename the method to `WriteDataToFile` and change data to be `char *` (unless there is a good reason you would want it to be vector of string).

Reply:
Thanks for your suggestions! Yes, this API is not good. Have changed it as you suggested. Also add the docs for `input_buffer_size` and `output_buffer_size`.